cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -277,21 +277,34 @@ func getSchedulerCommand(cfg *kubeadmapi.MasterConfiguration) []string {
 }
 
 // getProxyEnvVars builds a list of environment variables to use in the control plane containers in order to use the right proxy
-func getProxyEnvVars() []v1.EnvVar {
+func getProxyEnvVars(cfg *kubeadmapi.MasterConfiguration) []v1.EnvVar {
+	noProxyIPs := getNoProxyIPv4IPs(cfg)
+	var hasNoProxySet bool
+
 	envs := []v1.EnvVar{}
 	for _, env := range os.Environ() {
 		pos := strings.Index(env, "=")
 		if pos == -1 {
 			// malformed environment variable, skip it.
 			continue
 		}
-		name := env[:pos]
+		name := strings.ToLower(env[:pos])
 		value := env[pos+1:]
-		if strings.HasSuffix(strings.ToLower(name), "_proxy") && value != "" {
+
+		if strings.HasSuffix(name, "_proxy") && value != "" {
+			if name == "no_proxy" {
```

Comment:
you could also `continue` if name == no_proxy and set the env in 1 place rather than 2. WDYT? not a big issue IMO